### PR TITLE
release-19.2: fix server hangs on HTTP timeouts

### PR DESCRIPTION
**Pull request by knz**
Backport:
  * 1/1 commits from "*: new `httputil.Client` and methods, & use them" (#42536)
  * 1/1 commits from "cli/interactive_tests: opt out of telemetry by default" (#42537)
  * 1/1 commits from "lint: only run TestRoachLint nightly" (#42595)
  * 1/1 commits from "lint: properly parallelize TestVet/TestVet" (#42605) 

Please see individual PRs for details.

/cc @cockroachdb/release

### CODE REVIEW

**Comment by tbg** [comment]
LGTM but I'll defer to @bdarnell about postponing this for a few days until after 19.2.1.

**Review by bdarnell — APPROVED** [review]
LGTM. This should go into 19.2.2, but the commit for 19.2.1 has already been picked so I think this is ready to go after the master commit is merged.

**Pull request merged**
